torch/fx/experimental/symbolic_shapes.py

```diff
@@ -5771,6 +5772,25 @@ def simplify(self, expr: _SympyT) -> _SympyT:
         """Use known constraints and replacements to simplify the given expr"""
         expr = safe_expand(expr)
         expr = self.replace(expr)
+
+        if expr.has(Max):
+            max_replacements = {}
+            for atom in expr.atoms(Max):
+                a, b = atom.args
+                if b == 1 or b == 0:
+                    a, b = b, a
+                if a == 1 or a == 0:
+                    if (
+                        isinstance(b, Add)
```

Comment:
yeh very customized on both the 0/1 up there and the a+b here. 
the issue with landing changes like this, is that if users saw this working then will assume it generalize but it wont. its kind of a branchy behaviour. 

is the user allowed to do something like
torch._check(torch.sym_max(1, u0 + u1) == 20) to go around this issue